python/src/scippneutron/file_loading/_log_data.py

```diff
@@ -59,39 +51,33 @@ def _correct_nxlog_times(raw_times: sc.Variable,
             not provided, defaults to 1 (a no-op scaling factor).
     """
     try:
-        raw_times_s = sc.to_unit(raw_times, sc.units.s)
+        raw_times_ns = sc.to_unit(raw_times, sc.units.ns)
```

Comment:
```suggestion
        raw_times_ns = sc.to_unit(raw_times, sc.units.ns, copy=False)
```
to potentially avoid redundant copy.

But given that below we multiply be a `float64` scale, why is this converted at all?

Reply:
The scale and the units are independent of each other I think... from the nexus standard:

> The scaling_factor, when present, has to be applied to the time values in order to arrive at the units specified in the units attribute

So we need to do both conversions (changing unit and applying scaling factor) to arrive at a consistent unit of `ns`.

Unless I have misunderstood your comment?

Reply:
No, my bad, since I didn't realize that the other was `astype`, not `to_unit`, ...